db/blip_sync_context.go

```
@@ -870,3 +847,64 @@ func (bsc *BlipSyncContext) useHLV() bool {
 func (bsc *BlipSyncContext) sendRevTreeProperty() bool {
 	return bsc.activeCBMobileSubprotocol >= CBMobileReplicationV4 && bsc.clientType == BLIPClientTypeSGR2
 }
+
+// getKnownRevs will return deltaSrcRev (if delta sync is enabled), whether the known rev array represents a legacy
+// document, the parsed known revs in a map and error if unknown format is received.
+func (bsc *BlipSyncContext) getKnownRevs(ctx context.Context, docID string, knownRevsArray []interface{}) (deltaSrcRev string, changeIsLegacyRev bool, knownRevs map[string]bool, err error) {
+	knownRevs = make(map[string]bool)
+	// The first element of the knownRevsArray returned from CBL is the parent revision to use as deltaSrc for
+	// revtree clients. For HLV clients, use the cv as deltaSrc
+	if bsc.useDeltas && len(knownRevsArray) > 0 {
+		if revID, ok := knownRevsArray[0].(string); ok {
+			if bsc.useHLV() {
+				// extract cv from the known revs array
+				msgHLV, _, err := extractHLVFromBlipString(revID)
+				if err != nil {
+					base.DebugfCtx(ctx, base.KeySync, "Invalid known rev format for hlv on doc: %s falling back to full body replication.", base.UD(docID))
```

we are swallowing the error here, so we don't have much chance of ever debugging this

---

I don't think we want to error here fully given it can fall back to full body replication. Its possible that the first elem may be a revID (if the change response for this doc is a legacy rev) in this case we wouldn't want to error but maybe log to ensure its clear why its not being sent as a delta. 

I have added the error message and the revID string to the log line though so we can see why we fall into this.